src/exercise/06.js

```diff
@@ -14,18 +14,47 @@ function UsernameForm({onSubmitUsername}) {
   // you prefer from the options mentioned in the instructions)
   // 💰 For example: event.target.elements[0].value
   // 🐨 Call `onSubmitUsername` with the value of the input
+  const [error, setError] = useState(null)
+  const [IsLowercase, SetIsLowercase] = useState(false)
+  const [username, setUsername] = useState()
+  const userInputRef = useRef(null)
+  const handleSubmit = event => {
+    event.preventDefault()
+    
+    // onSubmitUsername(userInputRef.current.value)
 
+    // Extra 3
+    onSubmitUsername(username)
+  }
+  const HandleChange = e => {
+    // Extra 2
+    // const {value} = e.target // RQ : {value}=e.target =! value=e.target
+    // value === value.toLowerCase() ? SetIsLowercase(true) : SetIsLowercase(false)
+    // setError(IsLowercase ? null : 'Votre username doit etre en miniscule')
+    setUsername(e.target.value.toLowerCase())
+  }
   // 🐨 add the onSubmit handler to the <form> below
 
   // 🐨 make sure to associate the label to the input.
   // to do so, set the value of 'htmlFor' prop of the label to the id of input
   return (
-    <form>
+    <form onSubmit={handleSubmit}>
       <div>
         <label>Username:</label>
-        <input type="text" />
+        <input
+          type="text"
+          name="nameInput"
+          onChange={HandleChange}
```

Comment:
```suggestion
          onChange={handleChange}
```

just pour info les fonction normal, on peut les appeler dump functions commencent toujours par minuscule